src/entry.js

```diff
@@ -12,6 +13,10 @@ function StacheEntryPlugin() {
         require('./template-reference-variable')
       ];
 
+      if (!stacheJsonDataService.getStacheDataObject()) {
```

Comment:
This is more a check to make sure the data object doesn't already exist before trying to set it.   This way the data object only gets set (all the files converted) once when the plugin starts, instead of every time a file gets passed through the plugins.  Without this check (or one similar) every file passing through the plugins would cause the stache object to build again.  Before we were only building it once during the `app-extras.module`,  I could expose another method to throw a boolean if the data object is already there or not, but I figured the getData would suffice as it would either return a data object or undefined.